fastMatcher.go

```diff
@@ -561,11 +562,25 @@ func (m *FastMatcher) matchExec(token tokenType, tokenData []byte, tokenDataLen
 				return err
 			}
 
+			if m.buckets.IsResolved(0) {
+				return nil
+			}
+		}
+		objEndPos := m.tokens.Position()
+
+		objFastVal := NewObjectFastVal(m.tokens.data[objStartPos:objEndPos])
```

Comment:
You should defer the creation of the fastval until after you've confirmed that there are Ops to process, otherwise you will be incurring a significant parse cost on every document.  Surprised this wasn't caught by performance regression tests.